src/components/quantity-selector/quantity-selector.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from "classnames";
+import React, { PureComponent } from "react";
+import "../../styles/components/_quantity-selector.scss";
+import Button from "../button";
+import Icon from "../icon";
+
+export default class QuantitySelector extends PureComponent<IQuantitySelectorProps, any> {
+
+    public render() {
+        const count = this.props.count || 0;
+
+        if (count === 0) {
+            return <Button fluid variant="secondary" onClick={this.props.onIncrementClick} size="xsmall">
+                {this.props.buttonText}
+            </Button>;
+        }
+
+        return (
+            <div className="q-qs-main">
+                <div className="q-qs-left">
+                    {this.renderDecrementButton()}
+                </div>
+                <div className="q-qs-middle">
+                    {count}
+                </div>
+                <div className="q-qs-right">
+                    {this.renderIncrementButton()}
+                </div>
+            </div>
+        );
+    }
+    private renderDecrementButton() {
+        const deleteIcon = classNames("icon-trash");
+        const minusIcon = classNames("icon-minus");
+
+        if (this.props.count === 1) {
+            return <Icon
+                className={deleteIcon}
+                onClick={this.props.onDecrementClick}
+                name="delete"
+            />;
+        } else if (this.props.count && this.props.count > 1) {
```

Comment:
`this.props.count &&` this parts unnecessary

when count prop not exist `this.props.count > 0` returns false because relational comparison checks value exist or not exist.
try this on your console.
`const props={}; console.log(props.count > 0)`